tests/test_repocard.py

```diff
@@ -159,3 +173,142 @@ def test_metadata_eval_result(self):
         metadata_save(filepath, data)
         content = filepath.read_text().splitlines()
         self.assertEqual(content, DUMMY_MODELCARD_EVAL_RESULT.splitlines())
+
+
+class RepocardUpdateTest(unittest.TestCase):
+    _api = HfApi(endpoint=ENDPOINT_STAGING)
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Share this valid token in all tests below.
+        """
+        cls._token = TOKEN
+        cls._api.set_access_token(TOKEN)
+
+    @retry_endpoint
+    def setUp(self) -> None:
+        if os.path.exists(REPO_NAME):
+            shutil.rmtree(REPO_NAME, onerror=set_write_permission_and_retry)
+        logger.info(f"Does {REPO_NAME} exist: {os.path.exists(REPO_NAME)}")
+        self.repo = Repository(
+            REPO_NAME,
+            clone_from=f"{USER}/{REPO_NAME}",
+            use_auth_token=self._token,
+            git_user="ci",
+            git_email="ci@dummy.com",
+        )
+
+        with self.repo.commit("Add README to main branch"):
+            with open("README.md", "w+") as f:
+                f.write(DUMMY_MODELCARD_EVAL_RESULT)
+
+        self.existing_metadata = yaml.safe_load(
+            DUMMY_MODELCARD_EVAL_RESULT.strip().strip("---")
+        )
+
+    def tearDown(self) -> None:
+        self._api.delete_repo(repo_id=f"{REPO_NAME}", token=self._token)  # {USER}/
+        shutil.rmtree(REPO_NAME)
+
+    def test_update_dataset_name(self):
+        new_datasets_data = {"datasets": "['test/test_dataset']"}
+        metadata_update(f"{USER}/{REPO_NAME}", new_datasets_data, token=self._token)
+
+        self.repo.git_pull()
+        updated_metadata = metadata_load(f"{REPO_NAME}/README.md")
+        expected_metadata = self.existing_metadata.copy()
```

Comment:
```suggestion
        expected_metadata = copy.deepcopy(self.existing_metadata)
```

to copy nested objects in a dict. Otherwise:

```py
>>> a = {1: {2: 3}}
>>> b = a.copy()
>>> b[1][2] = 4
>>> b
{1: {2: 4}}
>>> a
{1: {2: 4}}
```